### PR TITLE
Prevent 'add a property' button from appearing over the 'allow multiple' menu

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR fixes a visual bug in the entity type editor where the "add a property" button appears on top of the "Allow multiple" menu.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204274556967336/1204521839453643/f) _(internal)_

## 🔍 What does this change?

The "add a property" button table cell has `zIndex: theme.zIndex.drawer + 1` so in order for the "allow multiple" menu to appear on top its zIndex has ben set to `theme.zIndex.drawer + 2`

### 🚢 Has this modified a publishable library?

This PR:

- [x] modifies an **npm**-publishable library and **I have added a changeset file(s)**

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

 - [x] does not affect the execution graph

## 📹 Demo

Before:
image

After:
image

### CODE REVIEW

File: .changeset/config.json [hunk 1]
```diff
@@ -12,8 +12,7 @@
     "@local/*",
     "@tests/*",
     "error-stack",
-    "deer",
-    "antsi"
+    "deer"
```
Comment (yusufkinatas) [inline .changeset/config.json:15]: is this change related with this PR?

Comment (luisbettencourt) [reply .changeset/config.json:15]: Sorry, it was discussed [here ](https://hashintel.slack.com/archives/C022217GAHF/p1683393353949689)

**Review by yusufkinatas — APPROVED** [review]
